packages/core/admin/admin/src/hooks/useSettingsForm/tests/index.test.js

```diff
@@ -0,0 +1,125 @@
+import * as React from 'react';
+import { setupServer } from 'msw/node';
+import { rest } from 'msw';
+import { renderHook, act } from '@testing-library/react-hooks';
+import { IntlProvider } from 'react-intl';
+import omit from 'lodash/omit';
+import useSettingsForm from '../index';
+
+const toggleNotification = jest.fn();
+
+jest.mock('@strapi/helper-plugin', () => ({
+  ...jest.requireActual('@strapi/helper-plugin'),
+  useNotification: jest.fn().mockImplementation(() => toggleNotification),
+  useOverlayBlocker: () => ({ lockApp: jest.fn(), unlockApp: jest.fn() }),
+}));
+
+jest.mock('../../../utils', () => ({
+  ...jest.requireActual('../../../utils'),
+  checkFormValidity: () => (null),
+}));
+
+jest.mock('lodash/omit');
```

Comment:
Why?

Reply:
Because it is the only way I found to test this part of my code 
`if (cleanedData.ssoLockedRoles) {
          cleanedData.ssoLockedRoles = [...new Set(cleanedData.ssoLockedRoles)];
        }`
inside the useSettingsForm